subsys/net/lib/sockets/sockets.c

```diff
@@ -45,8 +45,36 @@ const struct socket_op_vtable sock_fd_op_vtable;
 static inline void *get_sock_vtable(
 			int sock, const struct socket_op_vtable **vtable)
 {
-	return z_get_fd_obj_and_vtable(sock,
-				       (const struct fd_op_vtable **)vtable);
+	void *ctx;
+
+	ctx = z_get_fd_obj_and_vtable(sock,
+				      (const struct fd_op_vtable **)vtable);
+
+#ifdef CONFIG_USERSPACE
+	if (z_is_in_user_syscall()) {
```

Comment:
Should we check here that ctx is not null, so `if (ctx && z_is_in_user_syscall()) {`

Reply:
z_object_find() returns NULL if you pass it NULL (or any address that isn't a kernel object). assuming the success case is far more likely, I think it's slightly more efficient to omit a check at this function's level.